.github/workflows/ci_nightly.yml

```diff
@@ -50,24 +50,41 @@ jobs:
         shell: bash
         run: cargo xtask ci ${{ matrix.device }} --toolchain nightly
 
-      - if: failure()
-        name: Create or Update GitHub Issue
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # Check for unused dependencies through esp-hal packages
+      - name: Install and run cargo machete
         run: |
+          cargo install cargo-machete
+          cargo machete
+
+  check-global-symbols:
+    name: Check Global Symbols in esp-hal
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+       # Install the Rust toolchain for Xtensa devices:
+      - uses: esp-rs/xtensa-toolchain@v1.6
+        with:
+          version: 1.89.0.0
```

Comment:
Now that https://github.com/esp-rs/esp-hal/pull/4182 is in the queue, you'll need to change this to 1.90.0.0